docs/source/common/optimizers.rst

```diff
@@ -3,238 +3,322 @@
 
 .. tip:: ``self.optimizers()`` will return ``LightningOptimizer`` objects. You can access your own optimizer with ``optimizer.optimizer``. However, if you use your own optimizer to perform a step, Lightning won't be able to support accelerators and precision for you.
 
-.. code-block:: python
+.. tip:: Be careful where you call ``zero_grad`` or your model won't converge. It is good pratice to call ``zero_grad`` before ``manual_backward``.
 
-    def __init__(self):
-        self.automatic_optimization = False
 
-    def training_step(self, batch, batch_idx):
-        opt = self.optimizers()
+Learning rate scheduling
+------------------------
+From 1.3, Lightning supports learning rate scheduling in manual optimization.
+``lr_scheduler.step()`` is disabled in manual optimization so that you can call it at arbitrary intervals. Use ``self.lr_schedulers()`` in LightningModule to access your learning rate schedulers defined in ``LightningModule.configure_optimizers()``.
 
-        loss = self.compute_loss(batch)
-        self.manual_backward(loss)
+.. deprecated:: Before 1.3, ``lr_scheduler.step`` was called automatically in both manual and automatic optimization. 
 
-        # accumulate gradient batches
-        if batch_idx % 2 == 0:
-            opt.step()
-            opt.zero_grad()
+.. warning:: The learning rate scheduler keys are ignored in manual optimization.
 
-.. tip:: It is a good practice to provide the optimizer with a ``closure`` function that performs a ``forward``, ``zero_grad`` and ``backward`` of your model. It is optional for most optimizers, but makes your code compatible if you switch to an optimizer which requires a closure. See also `the PyTorch docs <https://pytorch.org/docs/stable/optim.html#optimizer-step-closure>`_.
+.. testcode:: python
+
+   # step every batch
+
+   def __init__(self):
+       super().__init__()
+       self.automatic_optimization = False
+
+   def training_step(self, batch, batch_idx):
+       # do foward, backward, and optimization
+       ...
 
-Here is the same example as above using a ``closure``.
+       # single scheduler
+       sch = self.lr_schedulers()
+       sch.step()
+
+       # multiple schedulers
+       sch1, sch2 = self.lr_schedulers()
+       sch1.step()
+       sch2.step()
 
 .. testcode:: python
 
-    def __init__(self):
-        self.automatic_optimization = False
+   def __init__(self):
+       super().__init__()
+       self.automatic_optimization = False
 
-    def training_step(self, batch, batch_idx):
-        opt = self.optimizers()
+   def training_step(self, batch, batch_idx):
+       # do forward, backward, and optimization
+       ...
 
-        def closure():
-            # Only zero_grad on the first batch to accumulate gradients
-            is_first_batch_to_accumulate = batch_idx % 2 == 0
-            if is_first_batch_to_accumulate:
-                opt.zero_grad()
```

Comment:
When using `LBFGS`-like optimizers which require a closure function to reevaluate the model, the closure always has to run `zero_grad` to make them work properly, so "gradient accumulation + closure" is not supported in PL as well as PyTorch, I think.